### PR TITLE
chore(test): Automation of Kubernetes e2e tests

### LINKED ISSUE #10001: Automation of the test scenarios for Secrets, ConfigMaps, Nodes and PVCs
Please make sure that E2E test cases are automated and run regularly on Windows and Mac platform and Linux.

### PR DESCRIPTION
**Pull request by amisskii**
Signed-off-by: Anton Misskii <amisskii@redhat.com>

### What does this PR do?

Modifies Kubernetes tests to run on the Linux platform in GitHub Actions, introduces a workflow to run only Kubernetes tests, and configures the e2e-main workflow to run all test suites using the test:e2e:all npm target.

### What issues does this PR fix or reference?

https://github.com/podman-desktop/podman-desktop/issues/10001

### How to test this PR?

- [ ] Tests are covering the bug fix or the new feature

### CODE REVIEW

File: .github/workflows/e2e-main.yaml [hunk 1]
```diff
@@ -33,6 +33,11 @@ on:
         description: 'Podman Desktop repository name'
         type: string
         required: true
+      npm_target:
+        default: 'test:e2e'
+        description: 'npm target to run tests'
```
Comment (jeffmaury) [inline .github/workflows/e2e-main.yaml]: Maybe extend description to include other values for the target ?

File: tests/playwright/src/specs/deploy-to-kubernetes.spec.ts [hunk 2]
```diff
@@ -55,27 +56,31 @@ test.beforeAll(async ({ runner, welcomePage, page, navigationBar }) => {
     await settingsBar.cliToolsTab.click();
 
     await ensureCliInstalled(page, 'Kind');
+    await createKindCluster(page, CLUSTER_NAME, true, CLUSTER_CREATION_TIMEOUT);
+  }
+  if (process.env.GITHUB_ACTIONS && process.env.RUNNER_OS === 'Linux') {
```
Comment (jeffmaury) [inline tests/playwright/src/specs/deploy-to-kubernetes.spec.ts]: Why can't we run it on our laptop ?

Comment (amisskii) [reply tests/playwright/src/specs/deploy-to-kubernetes.spec.ts]: The Kubernetes context validation in the beforeAll hook is applicable only for GitHub Actions (GHA) running on Linux. In this environment, a kind cluster is created before the tests are executed, so we ensure that the correct context is being used.

When running the tests locally or on GHA for other platforms, the kind cluster is created during the test execution. After the cluster is created, an assertion verifies that the appropriate Kubernetes context is used: https://github.com/podman-desktop/podman-desktop/blob/87cd8cada63c1a3daf49aa57efb14cf420a81958/tests/playwright/src/utility/operations.ts#L362

**Comment by amisskii** [comment]
test workflow run: https://github.com/amisskii/podman-desktop/actions/runs/12397338079

**Comment by amisskii** [comment]
test workflow run: https://github.com/amisskii/podman-desktop/actions/runs/12409039313

**Review by cbr7 — APPROVED** [review]

**Review by odockal — APPROVED** [review]

**Pull request merged**